tests/integration/pilot/main_test.go

```diff
@@ -103,10 +103,13 @@ func TestMain(m *testing.M) {
 			}
 			return ctx.Config().ApplyYAML("", string(crd))
 		}).
-		Setup(istio.Setup(&i, func(cfg *istio.Config) {
+		Setup(istio.Setup(&i, func(ctx resource.Context, cfg *istio.Config) {
 			cfg.Values["telemetry.v2.metadataExchange.wasmEnabled"] = "false"
 			cfg.Values["telemetry.v2.prometheus.wasmEnabled"] = "false"
-			cfg.Values["meshConfig.defaultConfig.proxyMetadata.PROXY_XDS_VIA_AGENT"] = "enable"
+			if !ctx.Clusters().IsMulticluster() {
+				// TODO make this compatible with multicluster
```

Comment:
TODO(stevenctl):

Reply:
also might worth commenting on why we can't set this field when it's not running as multi cluster mode?